sympy/printing/lambdarepr.py

```diff
@@ -108,6 +108,17 @@ def _print_MatMul(self, expr):
         "Matrix multiplication printer"
         return '({0})'.format(').dot('.join(self._print(i) for i in expr.args))
 
+    def _print_DotProduct(self, expr):
+        # DotProduct allows any shape order, but numpy.dot does matrix
+        # multiplication, so we have to make sure it gets 1 x n by n x 1.
+        arg1, arg2 = expr.args
+        if arg1.shape[0] != 1:
```

Comment:
It might be good to check that len(arg1.shape) == len(arg2.shape) == 2 here. I think that should always be the case for a `Matrix` object, but we could have an N-dimensional array object in the future, right?